File: parking_lot_system_go/parkinglot/observer.go
package parkinglot

import "fmt"

type Observer interface {
	Update(message string)
}

type ParkingLotObserver struct{}

func (o *ParkingLotObserver) Update(message string) {
	fmt.Println("Notification:", message)
}

type Subject struct {
	observers []Observer
}

func (s *Subject) AddObserver(observer Observer) {
	s.observers = append(s.observers, observer)
}

func (s *Subject) RemoveObserver(observer Observer) {
	for i, obs := range s.observers {
		if obs == observer {
			s.observers = append(s.observers[:i], s.observers[i+1:]...)
			break
		}
	}
}

func (s *Subject) NotifyObservers(message string) {
	for _, observer := range s.observers {
		observer.Update(message)
	}
}

File: parking_lot_system/parking_lot/observer.py
class Observer:
    def update(self, message):
        raise NotImplementedError

class ParkingLotObserver(Observer):
    def update(self, message):
        print(f"Notification: {message}")

class Subject:
    def __init__(self):
        self._observers = []

    def add_observer(self, observer):
        self._observers.append(observer)

    def remove_observer(self, observer):
        self._observers.remove(observer)

    def notify_observers(self, message):
        for observer in self._observers:
            observer.update(message)
